synapse/handlers/room.py

```diff
@@ -704,13 +704,56 @@ def _make_and_store_3pid_invite(
             token_id,
             txn_id
     ):
+        room_state = yield self.hs.get_state_handler().get_current_state(room_id)
+
+        inviter_display_name = ""
+        inviter_avatar_url = ""
+        member_event = room_state.get((EventTypes.Member, user.to_string()))
+        if member_event:
+            if (
+                    "displayname" in member_event.content and
+                    member_event.content.get("displayname", "")
+            ):
+                inviter_display_name = member_event.content["displayname"]
+            if (
+                    "avatar_url" in member_event.content and
+                    member_event.content.get("avatar_url", "")
+            ):
+                inviter_avatar_url = member_event.content["avatar_url"]
+
+        canonical_room_alias = ""
+        canonical_alias_event = room_state.get((EventTypes.CanonicalAlias, ""))
+        if canonical_alias_event and canonical_alias_event.content.get("alias", ""):
+            canonical_room_alias = canonical_alias_event.content["alias"]
+
+        room_name = ""
+        room_name_event = room_state.get((EventTypes.Name, ""))
+        if room_name_event and room_name_event.content.get("name", ""):
+            room_name = room_name_event.content["name"]
+
+        room_join_rules = ""
+        join_rules_event = room_state.get((EventTypes.JoinRules, ""))
+        if join_rules_event and join_rules_event.content.get("join_rule", ""):
+            room_join_rules = join_rules_event.content["join_rule"]
```

Comment:
You could do:

    room_join_rules = join_rules_event.content.get("join_rule", "")